nectar/src/command/trade/event_loop.rs

```diff
@@ -87,97 +87,95 @@ impl EventLoop {
+                    if let Some(new_btc_balance) = new_btc_balance {
+                        match new_btc_balance {
+                            Ok(new_btc_balance) => {
+                                let _ = self.handle_btc_balance_update(new_btc_balance).map_err(|err|
+                                    tracing::error!("BTC balance update handing failed: {:#}", err));
+                            }
+                            Err(err) => tracing::error!("BTC balance update failed: {:#}", err),
+                        }
+                    }
                 },
-                dai_balance_update = self.dai_balance_update_receiver.next().fuse() => {
-                    self.handle_dai_balance_update(dai_balance_update.unwrap());
+                new_dai_balance = self.dai_balance_update_receiver.next().fuse() => {
+                    if let Some(new_dai_balance) = new_dai_balance {
+                        match new_dai_balance {
+                            Ok(new_dai_balance) => {
+                                let _ = self.handle_dai_balance_update(new_dai_balance).map_err(|err|
+                                    tracing::error!("Dai balance update handing failed: {:#}", err));
+                            }
+                            Err(err) => tracing::error!("Dai balance update failed: {:#}", err),
+                        }
+                    }
                 }
             }
         }
     }
 
-    fn handle_rate_update(&mut self, rate_update: Result<MidMarketRate>) {
-        match rate_update {
-            Ok(new_rate) => {
-                let result = self.maker.update_rate(new_rate);
-                match result {
-                    Ok(Some(PublishOrders {
-                        new_sell_order,
-                        new_buy_order,
-                    })) => {
-                        self.swarm.orderbook.publish(
-                            new_sell_order.to_comit_order(self.maker.swap_protocol(Position::Sell)),
-                        );
-                        self.swarm.orderbook.publish(
-                            new_buy_order.to_comit_order(self.maker.swap_protocol(Position::Buy)),
-                        );
-                        self.swarm.orderbook.clear_own_orders();
-                    }
-
-                    Ok(None) => (),
-                    Err(e) => tracing::warn!("Rate update yielded error: {}", e),
-                }
-            }
-            Err(e) => {
-                self.maker.invalidate_rate();
-                tracing::error!(
-                    "Unable to fetch latest rate! Fetching rate yielded error: {}",
-                    e
-                );
-            }
+    fn handle_rate_update(&mut self, new_rate: MidMarketRate) -> Result<()> {
+        let publish_order = self
+            .maker
+            .update_rate(new_rate)
+            .context("Unable to fetch latest rate! Fetching rate yielded error")?;
```

Comment:
I think there is a copy-pasta error here! That error used to be emitted if an `Err` was passed to the function but that now happens outside.

Reply:
I think the same applies for all other functions as well.